src/app/src/main/kotlin/org/imperial/mrc/hint/db/CalibrateDataRepository.kt

```diff
@@ -0,0 +1,89 @@
+package org.imperial.mrc.hint.db
+
+import org.jooq.tools.json.JSONArray
+import org.jooq.tools.json.JSONObject
+import org.springframework.stereotype.Component
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.imperial.mrc.hint.exceptions.CalibrateDataException
+import org.imperial.mrc.hint.models.CalibrateResultRow
+import java.sql.Connection
+import java.sql.DriverManager
+import java.sql.Statement
+import java.util.Properties
+import java.util.Locale
+import java.sql.ResultSet
+import java.sql.SQLException
+
+const val DEFAULT_QUERY = """SELECT
+age_group,area_id,
+calendar_quarter,
+indicator,
+ROUND(lower, 4) AS lower,
+ROUND(mean, 4) AS mean,
+ROUND(mode, 4) AS mode,
+sex,
+ROUND(upper, 4) AS upper
+FROM data"""
+
+interface CalibrateDataRepository
+{
+    fun getDataFromPath(path: String): JSONArray
+}
+
+@Component
+class JooqCalibrateDataRepository: CalibrateDataRepository
+{
+
+    private fun convertToArrayList(resultSet: ResultSet): ArrayList<CalibrateResultRow> {
+        val list: ArrayList<CalibrateResultRow> = arrayListOf()
+        while (resultSet.next()) {
+            val row = CalibrateResultRow(
+                resultSet.getString("indicator"),
+                resultSet.getString("calendar_quarter"),
+                resultSet.getString("age_group"),
+                resultSet.getString("sex"),
+                resultSet.getString("area_id"),
+                resultSet.getFloat("mode"),
+                resultSet.getFloat("mean"),
+                resultSet.getFloat("lower"),
+                resultSet.getFloat("upper")
+            )
+            list.add(row)
+        }
+        return list
```

Comment:
```suggestion
               resultSet.use {
            generateSequence {
                if (it.next()) {
                    CalibrateResultRow(
                        it.getString("indicator"),
                        it.getString("calendar_quarter"),
                        it.getString("age_group"),
                        it.getString("sex"),
                        it.getString("area_id"),
                        it.getFloat("mode"),
                        it.getFloat("mean"),
                        it.getFloat("lower"),
                        it.getFloat("upper")
                    )
                } else {
                    null
                }
            }.toList()
        }
```


I think your implementation is replicating JAVA but you could make it more efficient with .use {}